### DISCUSSION TITLE
Avalara Plugin calculates wrong taxes for discounted product (it ask more money for these products)

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#9682] [General] [answered]
Hi,
we have a problem with taxes (Avalara Plugin) 
because there are not calculated in the right way for discounted products.

This report is from one of our customer in USA.

This is an example:

Jersey retail $100
Tax 10%
Promocode $25

The customer buy a jersey for 100$ using the promocode 25$

Currently the plugin add 10% to the jersey
100+10% = $110
And after it remove $25 

The total is $85.
The customer paid more tax than it should be.

We should apply 10% of tax on the discounted value $75: 
$100-$25 = $75
$75 + 10% = $82.50

Currently customer pay $85 but he sould pay $82.50.
We are asking more money (+$2.50)

Let me know If there are some setting I miss or if it's a bug and it could be fixed.

Thanks in advance and kind regards,

Davide

**Comment by karolkielecki** [comment]
@DidoMarchet Is that Saleor 3.1? Could you please provide an API response?

**Reply by DidoMarchet** [reply]
Hi @karolkielecki ,
this is the response (if you need another kind of response let me know):

```json
[
   {
      "data":{
         "checkout":{
            "id":"Q2hlY2tvdXQ6YjFlNTA0ZDctZGM5Ny00Yzg0LTg1MGYtOTM1Nzg4ODEyNzA3",
            "shippingPrice":{
               "gross":{
                  "amount":6.95,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "net":{
                  "amount":6.95,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "tax":{
                  "amount":0.0,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "__typename":"TaxedMoney"
            },
            "subtotalPrice":{
               "gross":{
                  "amount":97.98,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "net":{
                  "amount":89.99,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "tax":{
                  "amount":7.99,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "__typename":"TaxedMoney"
            },
            "discount":{
               "amount":50.0,
               "currency":"USD",
               "__typename":"Money"
            },
            "discountName":null,
            "totalPrice":{
               "gross":{
                  "amount":54.93,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "net":{
                  "amount":46.94,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "tax":{
                  "amount":7.99,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "__typename":"TaxedMoney"
            },
            "voucherCode":"TEST",
            "giftCards":[
               
            ],
            "__typename":"Checkout"
         }
      },
      "extensions":{
         "cost":{
            "requestedQueryCost":1,
            "maximumAvailable":50000
         }
      }
   }
]
```

As you can see in:
```json
"totalPrice":{
               "gross":{
                  "amount":54.93,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "net":{
                  "amount":46.94,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "tax":{
                  "amount":7.99,
                  "currency":"USD",
                  "__typename":"Money"
               },
               "__typename":"TaxedMoney"
            },
```
tax is calculated on the undiscounted price.

Thanks and kind regards,

Davide

**Comment by DidoMarchet** [comment]
Hi @karolkielecki 
in the screen below a screenshot of Avalara.
The order is inserted with the full price and without sales/discount.

![photo_2022-05-04_17-38-26](https://user-images.githubusercontent.com/38940340/166718366-d897bc65-cdac-4103-be61-843265e8c604.jpg)


But the product is in sale

<img width="511" alt="Screenshot 2022-05-04 at 17 39 08" src="https://user-images.githubusercontent.com/38940340/166718457-9f99ce6b-3f6b-4e67-9397-dd45e668db62.png">


that mean the taxes payed is for the whole order and no sales or discount attached.

Let me know,
thanks and kind regards,

Davide

**Comment by DidoMarchet** [comment]
Hi @karolkielecki ,
do you have any update ?

Thanks and kind regards,
Davide

**Comment by timuric** [comment]
@DidoMarchet which Saleor version are you running?

**Comment by timuric** [comment, marked as answer]
@DidoMarchet this issue was already patched here, and will be released soon in 3.4 https://github.com/saleor/saleor/pull/9633

**Reply by DidoMarchet** [reply]
Hi @timuric !
We are waiting for this fix.
Thanks and kind regards,

Davide
